tests/unit/test_model_setup.py

```diff
@@ -0,0 +1,918 @@
+    # -----> Reference scalar
+    reference_scalar = [np.ones(2), 2.0 * np.ones(2), 100.0 * np.ones(2)]
+
+    # -----> Populating list of model parameters
+    model_params = []
+    before_runs = [True, False]
+    for before_run in before_runs:
+        for case in range(3):
+            model_param = {
+                "use_ad": True,
+                "case": case,
+                "before_run": before_run,
+                "mdg": mdg,
+                "box": box,
+                "time_step": time_steps[case],
+                "end_time": end_times[case],
+                "bc_flow_type_primary": bc_flow_type_primary[case],
+                "bc_flow_type_secondary": bc_flow_type_secondary[case],
+                "bc_flow_values_primary": bc_flow_values_primary[case],
+                "bc_flow_values_secondary": bc_flow_values_secondary[case],
+                "source_flow_primary": source_flow_primary[case],
+                "source_flow_secondary": source_flow_secondary[case],
+                "storativity_primary": storativity_primary[case],
+                "storativity_secondary": storativity_secondary[case],
+                "aperture_primary": aperture_primary[case],
+                "aperture_secondary": aperture_secondary[case],
+                "biot_alpha_primary": biot_alpha_primary[case],
+                "biot_alpha_secondary": biot_alpha_secondary[case],
+                "permeability_primary": permeability_primary[case],
+                "permeability_secondary": permeability_secondary[case],
+                "viscosity_primary": viscosity_primary[case],
+                "viscosity_secondary": viscosity_secondary[case],
+                "vector_source_primary": vector_source_primary[case],
+                "vector_source_secondary": vector_source_secondary[case],
+                "vector_source_interface": vector_source_interface[case],
+                "reference_scalar": reference_scalar[case],
+                "bc_mech_type_primary": bc_mech_type_primary[case],
+                "bc_mech_values_primary": bc_mech_values_primary[case],
+            }
+            model_params.append(model_param)
+
+    @pytest.mark.parametrize("model_params", model_params)
+    def test_non_default_parameters_contact_mechanics_biot(self, model_params):
+        """Test model with non-default scalar sources"""
+
+        # Construct the class
+        class NonDefaultContactMechanicsBiot(pp.ContactMechanicsBiot):
+            """Model with non-default parameters."""
+
+            def __init__(self, params):
+                super().__init__(params)
+
+            def create_grid(self) -> None:
+                """Modify default grid.
+
+                Create a Cartesian grid in a unit square with two cells in the horizontal
+                direction and one cell in the vertical direction. A single vertical fracture
+                is included in the middle of the domain.
+
+                """
+                self.mdg = self.params["mdg"]
```

Comment:
Where is the mdg created?

Reply:
`mdg` is created outside the test and then passed as a model parameter to the `NonDefaultContactMechanicsBiot` class. This is to avoid having to create the same `mdg` every time the class is instantiated.